packages/api/test/e2e/promise-metadata.spec.js

```diff
@@ -0,0 +1,68 @@
+    const genesisHash = '0x8579ded4438b26597e188cda8579fdd6abf09b74ab672e604b9584cd8b4c3d0b';
+    const specVersion = '62';
+    const prebundles = {};
+    const key = `${genesisHash}-${specVersion}`;
+    prebundles[key] = rpcData;
+    const api = await ApiPromise.create({
+      prebundles
+    });
+
+    expect(api.genesisHash).toBeDefined();
+    expect(api.runtimeMetadata).toBeDefined();
+    expect(api.runtimeVersion).toBeDefined();
+    expect(api.rpc).toBeDefined();
+    expect(api.query).toBeDefined();
+    expect(api.tx).toBeDefined();
+    expect(api.derive).toBeDefined();
+  });
+
+  it('Create API instance without metadata and makes the runtime, rpc, state & extrinsics available', async() => {
+    const prebundles = {};
+    const api = await ApiPromise.create({
+      prebundles
+    });
+
+    expect(api.genesisHash).toBeDefined();
+    expect(api.runtimeMetadata).toBeDefined();
+    expect(api.runtimeVersion).toBeDefined();
+    expect(api.rpc).toBeDefined();
+    expect(api.query).toBeDefined();
+    expect(api.tx).toBeDefined();
+    expect(api.derive).toBeDefined();
+  });
+
+  it('Create API instance with incorect metadata map and makes the runtime, rpc, state & extrinsics available', async() => {
+    const rpcData = 'invalid data';
+    const genesisHash = '0x8579ded4438b26597e188cda8579fdd6abf09b74ab672e604b9584cd8b4c3d0b';
+    const specVersion = '62';
+    const prebundles = {};
+    const key = `${genesis
```

Comment:
`expect api.runtimeMetadata.toHex()` to be rpcData (or: not to be 'invalid data')

Reply:
Metadata.toHex() operation is not supported..
![Screen Shot 2019-04-30 at 10 58 24 AM](https://user-images.githubusercontent.com/29415595/56932289-3babd100-6b37-11e9-838a-4390fbbd70a3.png)